benchmarks/apps/openmm/openmm-gpu-bench.py

```diff
@@ -0,0 +1,75 @@
+# Mangled together with the help of David Wright.
+
+import time
+from sys import stdout
+
+from simtk import unit
+from openmm import app
+import openmm as mm
+
+
+def time_msg(msg):
+    "Utility function to print some messages with a leading timestamp."
+    t = time.localtime(time.time())
+    print(f"{t.tm_year:02d}-{t.tm_mon:02d}-{t.tm_mday:02d}T{t.tm_hour:02d}:"
+          f"{t.tm_min:02d}:{t.tm_sec:02d} >>> {msg}")
+
+
+time_msg("Starting up...")
+
+params = app.CharmmParameterSet('top_all27_prot_lipid.rtf',
+                                'par_all27_prot_lipid.prm')
+psf = app.CharmmPsfFile('benchmark.psf')
+pdb = app.PDBFile('benchmark.pdb')
+
+coords = pdb.positions
+min_crds = [coords[0][0], coords[0][1], coords[0][2]]
+max_crds = [coords[0][0], coords[0][1], coords[0][2]]
+
+for coord in coords:
+    min_crds[0] = min(min_crds[0], coord[0])
+    min_crds[1] = min(min_crds[1], coord[1])
+    min_crds[2] = min(min_crds[2], coord[2])
+    max_crds[0] = max(max_crds[0], coord[0])
+    max_crds[1] = max(max_crds[1], coord[1])
+    max_crds[2] = max(max_crds[2], coord[2])
+
+psf.setBox(max_crds[0]-min_crds[0],
+           max_crds[1]-min_crds[1],
+           max_crds[2]-min_crds[2],
+           )
+
+system = psf.createSystem(params,
+                          nonbondedMethod=app.PME,
+                          nonbondedCutoff=0.8*unit.nanometers,
+                          constraints=app.HBonds,
+                          rigidWater=True,
+                          ewaldErrorTolerance=0.0005)
+time_msg('System creation complete')
+integrator = mm.LangevinIntegrator(300*unit.kelvin, 1.0/unit.picoseconds,
```

Comment:
you could replace the `LangevinIntegrator` with the [`LangevinMiddleIntegrator`](http://docs.openmm.org/latest/api-python/generated/openmm.openmm.LangevinMiddleIntegrator.html), which would allow you to use a 4 fs timestep through [hydrogen mass repartitioning](https://pubs.acs.org/doi/10.1021/ct5010406):
```python
integrator = mm.LangevinMiddleIntegrator(300*unit.kelvin, 1.0/unit.picoseconds, 4.0*femtoseconds)
```

Reply:
Uhm, with this change the simulation crashes:
```
Traceback (most recent call last):
  File "/mnt/lustre/tursafs1/home/do006/do006/dc-gior1/repo/excalibur-tests/benchmarks/apps/openmm/openmm-gpu-bench.py", line 74, in <module>
    simulation.step(10000)
  File "/mnt/lustre/tursafs1/home/do006/do006/dc-gior1/repo/excalibur-tests/benchmarks/spack/tursa/cpu/opt/spack/linux-rhel8-zen/gcc-8.4.1/openmm-7.7.0-pxgpqdw72wzizt4mocqfy3yzysfcqzd2/lib/python3.10/site-packages/openmm/app/simulation.py", line 141, in step
    self._simulate(endStep=self.currentStep+steps)
  File "/mnt/lustre/tursafs1/home/do006/do006/dc-gior1/repo/excalibur-tests/benchmarks/spack/tursa/cpu/opt/spack/linux-rhel8-zen/gcc-8.4.1/openmm-7.7.0-pxgpqdw72wzizt4mocqfy3yzysfcqzd2/lib/python3.10/site-packages/openmm/app/simulation.py", line 206, in _simulate
    self.integrator.step(10) # Only take 10 steps at a time, to give Python more chances to respond to a control-c.
  File "/mnt/lustre/tursafs1/home/do006/do006/dc-gior1/repo/excalibur-tests/benchmarks/spack/tursa/cpu/opt/spack/linux-rhel8-zen/gcc-8.4.1/openmm-7.7.0-pxgpqdw72wzizt4mocqfy3yzysfcqzd2/lib/python3.10/site-packages/openmm/openmm.py", line 16994, in step
    return _openmm.LangevinMiddleIntegrator_step(self, steps)
openmm.OpenMMException: Particle coordinate is nan
```

Reply:
ah, I think when creating the system you'd need to [set the `hydrogenMass`](http://docs.openmm.org/latest/userguide/application/02_running_sims.html?highlight=hydrogenmass#heavy-hydrogens) to something like 3 or 4 amu, but this is system dependent. And then you'd need to run a series of equilibration simulations, gradually incrementing the timestep. Perhaps it's not worth the time / effort given what you have is already working!